### PR TITLE
Bad Construction

### PR DESCRIPTION
**Pull request by Maithem**

### CODE REVIEW

**Comment by no2chem** [comment]
This is just a test PR to run against the perf framework, right?

**Comment by Maithem** [comment]
@no2chem Right. Also, note that since "this" is leaked outside the constructor, some of the memory guarantees  on final fields might not always be honored. In other words, some parts of the runtime need to be re-organized

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **8a49e72**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #621 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/HOcT9PvTAvSIZS5x5BojN67P2bpnk1oF)

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java [hunk 1]
```diff
@@ -28,11 +34,94 @@
...
+
+    public static MetricRegistry getMetrics() {
+        return CorfuRuntime.metrics;
+    }
+
+    public CorfuRuntimeParameters getParameters() {
+        return this.parameters;
+    }
+
+    public LayoutView getLayoutView() {
```
Comment (kspirov) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: Annotate with Lombok @Getter all fields you return with this and remove these methods.  - CorfuRuntime is big enough

Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: Hi @kspirov , I am aware of lambok notation. This was a **test** pull request to see if lombok's annotation were generating code that affects our performance. 

Apparently, lombok's lazy evaluation getters were masking object construction issues that appear to drop down performance to 1tx/sec

**Pull request closed without merging**